File: 191_NumberOf_1_Bits.py
class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        return bin(n).count('1')
        
class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        oriBin = "{0:032b}".format(n)
        return oriBin.count('1')
        

File: 191_NumberOf_1_Bits.cpp
class Solution {
public:
    int hammingWeight(uint32_t n) {
        int count(0);
		uint32_t mask(1);
		while(n){
			if(n & mask)
				count++;
			n >>= 1;
		}
		return count;
    }
};

//Better solution than mine
int hammingWeight(uint32_t n)
{
    int res = 0;
    while(n)
    {
        n &= n - 1;
        ++ res;
    }
    return res;
}